samples/star/src/main/kotlin/com/slack/circuit/star/data/Petfinder.kt

```diff
@@ -161,63 +162,67 @@ data class Pagination(
 )
 
 @JsonClass(generateAdapter = true)
-data class AuthenticationData(
+data class AuthenticationResponse(
   @Json(name = "token_type") val tokenType: String,
   @Json(name = "expires_in") val expiresIn: Long,
   @Json(name = "access_token") val accessToken: String,
 )
 
+suspend fun TokenStorage.updateAuthData(response: AuthenticationResponse) {
+  updateAuthData(
+    AuthenticationData(
+      response.tokenType,
+      Instant.now().plusSeconds(response.expiresIn),
+      response.accessToken
+    )
+  )
+}
+
 /**
  * A hypothetical token manager that stores an auth token. Just in-memory and not thread-safe for
  * now.
  */
-class TokenManager(private val api: PetfinderAuthApi) {
-  // TODO store this somewhere
-  private var expiration: Instant = Instant.EPOCH
-  private var tokenType: String? = null
-  private var token: String? = null
-
-  /** Returns a token if */
-  fun authenticate(request: Request): Request {
+class TokenManager(private val api: PetfinderAuthApi, private val tokenStorage: TokenStorage) {
+  private val mutex = Mutex()
+
+  /** Authenticates a [request]. */
+  suspend fun authenticate(request: Request): Request {
     return authenticate(request, false)
   }
 
-  private fun authenticate(request: Request, isAfterRefresh: Boolean): Request {
+  private suspend fun authenticate(request: Request, isAfterRefresh: Boolean): Request {
```

Comment:
why make it suspend now?

Reply:
Because the underlying datastore functions it uses are too